src/task_system/task_system_test/TaskSystemControllerTest.cs

```diff
@@ -430,5 +430,127 @@ public async Task DeleteTask_ReturnsBadRequest_WhenModelStateIsInvalid()
+                },
+                new TaskItem
+                {
+                    Id = 5,
+                    TaskType = TaskItemType.WakeUpCall,
+                    Description = "Provide a wake-up call at 6:00 AM for room 204.",
+                    RoomId = 204,
+                    RequesterId = 8,
+                    AssigneeId = 9,
+                    Status = TaskItemStatus.Completed,
+                    CreatedAt = new DateTime(2024, 10, 16, 6, 0, 0)
+                },
+                new TaskItem
+                {
+                    Id = 6,
+                    TaskType = TaskItemType.LaundryService,
+                    Description = "Pick up laundry from room 205 and deliver it back clean.",
+                    RoomId = 205,
+                    RequesterId = 10,
+                    AssigneeId = 11,
+                    Status = TaskItemStatus.InProgress,
+                    CreatedAt = new DateTime(2024, 10, 17, 10, 0, 0)
+                },
+                new TaskItem
+                {
+                    Id = 7,
+                    TaskType = TaskItemType.SpaAndMassage,
+                    Description = "Schedule a massage for the guest in room 206 at 3:00 PM.",
+                    RoomId = 206,
+                    RequesterId = 12,
+                    AssigneeId = 13,
+                    Status = TaskItemStatus.Pending,
+                    CreatedAt = new DateTime(2024, 10, 18, 15, 0, 0)
+                }
+            };
+
+            _mockRepo.Setup(r => r.GetTaskByAccountIdAsync(accountId)).ReturnsAsync(tasks);
+
+            var controllerContext = new ControllerContext
+            {
+                HttpContext = new DefaultHttpContext()
+            };
+            controllerContext.HttpContext.Request.Headers["X-API-Key"] = "valid-api-key";
+            _controller.ControllerContext = controllerContext;
+
+            var result = await _controller.GetTasksByAccountId(accountId);
+
+            var okResult = Assert.IsType<OkObjectResult>(result);
+            var response = Assert.IsType<TaskSystemResponse<IEnumerable<TaskItem>>>(okResult.Value);
+        }
+
+        [Fact]
+        public async Task GetTasksByAccountId_ReturnsUnauthorized_WhenApiKeyInvalid()
+        {
+            int accountId = 1;
+            _mockPermValidator.Setup(p => p.ValidateAccountId(accountId, It.IsAny<string>())).Returns(false);
+
+            var controllerContext = new ControllerContext
+            {
+                HttpContext = new DefaultHttpContext()
+            };
+            controllerContext.HttpContext.Request.Headers["X-API-Key"] = "invalid-api-key";
+            _controller.ControllerContext = controllerContext;
+
+            var result = await _controller.GetTasksByAccountId(accountId);
+
+            var unauthorizedResult = Assert.IsType<UnauthorizedObjectResult>(result);
+            var response = Assert.IsType<TaskSystemResponse<string>>(unauthorizedResult.Value);
+        }
```

Comment:
## Non-Blocking
### Missing test: when accountID != sessionKey.accountId
This can be done by using the _mockPermValidator object in the test class. 
See line 113 as an example: 
```_mockPermValidator.Setup(v => v.ValidatePermissions(PermissionNames.VIEW_TASKS, apiKey)).Returns(true);```